aws-auth-plugins-core/src/main/java/com/amplifyframework/auth/plugins/core/AWSCognitoIdentityPoolOperations.kt

```diff
@@ -0,0 +1,214 @@
+        with(awsCredentials) {
+            AWSCredentials.createAWSCredentials(accessKeyId, secretAccessKey, sessionToken, expiration)
+        }?.let {
+            AuthSessionResult.success(it)
+        } ?: AuthSessionResult.failure(UnknownException("Failed to fetch AWS credentials."))
+
+    suspend fun fetchAWSCognitoIdentityPoolDetails(
+        logins: List<LoginProvider>,
+        forceRefresh: Boolean
+    ): AWSCognitoIdentityPoolDetails {
+        logger.debug("fetchAWSCognitoIdentityPoolDetails: get cached AWS credentials")
+        val currentLoginProvider = deserializeLogins(awsAuthCredentialStore.get(KEY_LOGINS_PROVIDER))
+        val currentIdentityId = awsAuthCredentialStore.get(KEY_IDENTITY_ID)
+        val currentAWSCredentials = deserializeCredential(awsAuthCredentialStore.get(KEY_AWS_CREDENTIALS))
+
+        logger.debug("fetchAWSCognitoIdentityPoolDetails: start fetching identity id")
+        val identityIdResult = if (currentIdentityId == null || currentIdentityId.isBlank()) {
+            try {
+                val identityId = getIdentityId(logins)
+                AuthSessionResult.success(identityId)
+            } catch (exception: AuthException) {
+                AuthSessionResult.failure(exception)
+            }
+        } else {
+            AuthSessionResult.success(currentIdentityId)
+        }
+
+        val newLogin = logins != currentLoginProvider
+        logger.debug("fetchAWSCognitoIdentityPoolDetails: start fetching AWS credentials")
+        val awsCredentialsResult = if (currentAWSCredentials == null || !isValidSession(currentAWSCredentials) ||
+            newLogin || forceRefresh
+        ) {
+            when (identityIdResult.type) {
+                AuthSessionResult.Type.FAILURE -> AuthSessionResult.failure(identityIdResult.error)
+                AuthSessionResult.Type.SUCCESS -> {
+                    try {
+                        val (identityId, awsCredentials) = getAWSCredentials(identityIdResult.value, logins)
+                        awsAuthCredentialStore.put(KEY_LOGINS_PROVIDER, Json.encodeToString(logins))
+                        awsAuthCredentialStore.put(KEY_IDENTITY_ID, identityId ?: "")
+                        awsAuthCredentialStore.put(KEY_AWS_CREDENTIALS, Json.encodeToString(awsCredentials))
+                        logger.debug("fetchAWSCognitoIdentityPoolDetails: cached AWS credentials")
+                        getCredentialsResult(awsCredentials)
+                    } catch (exception: AuthException) {
+                        AuthSessionResult.failure(exception)
+                    }
+                }
+            }
+        } else {
+            getCredentialsResult(currentAWSCredentials)
+        }
+
+        return AWSCognitoIdentityPoolDetails(identityIdResult, awsCredentialsResult)
+    }
+
+    fun clearCredentials() {
+        logger.debug("clearCredentials: clear cached AWS credentials")
```

Comment:
Shouldnt it be verbose since you are logging a few things in this class?

Reply:
The debug logs will only be printed if Amplify Logging plugin is added with log level as debug.